src/dataflow/src/render/mod.rs

```diff
@@ -848,9 +910,15 @@ pub mod plan {
                     let input_mapper = JoinInputMapper::new(inputs);
 
                     // Plan each of the join inputs independently.
+                    // The `plans` get surfaced upwards, and the `input_keys` should
+                    // be used as part of join planning / to validate the existing
```

Comment:
+1 for doing join planning here.

Reply:
I don't think that is the right thing to do, without a fair bit of other work first.